charts/k8ssandra/templates/cassandra/cassdc.yaml

```diff
@@ -105,25 +112,41 @@ spec:
     spec:
       initContainers:
       - name: server-config-init
-     {{- if .Values.reaper.enabled }}
+     {{- if (or .Values.cassandra.auth.enabled .Values.reaper.enabled) }}
       - name: jmx-credentials
         image: busybox
         imagePullPolicy: IfNotPresent
         env:
-          - name: JMX_USERNAME
+        {{- if .Values.reaper.enabled }}
+          - name: REAPER_JMX_USERNAME
             valueFrom:
               secretKeyRef:
                 name: {{ .Release.Name }}-reaper-secret-k8ssandra
                 key: username
-          - name: JMX_PASSWORD
+          - name: REAPER_JMX_PASSWORD
             valueFrom:
               secretKeyRef:
                 name: {{ .Release.Name }}-reaper-secret-k8ssandra
                 key: password
+          {{- end }}
+          - name: SUPERUSER_JMX_USERNAME
+            valueFrom:
+              secretKeyRef:
+                name: {{ $superuserSecret }}
+                key: username
+          - name: SUPERUSER_JMX_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ $superuserSecret }}
+                key: password
         args:
           - /bin/sh
           - -c
-          - echo -n "$JMX_USERNAME $JMX_PASSWORD" > /config/jmxremote.password
+        {{- if .Values.reaper.enabled }}
+          - echo "$REAPER_JMX_USERNAME $REAPER_JMX_PASSWORD" > /config/jmxremote.password && echo "$SUPERUSER_JMX_USERNAME $SUPERUSER_JMX_PASSWORD" >> /config/jmxremote.password
+        {{- else }}
+          - echo echo "$SUPERUSER_JMX_USERNAME $SUPERUSER_JMX_PASSWORD" >> /config/jmxremote.password
```

Comment:
echo echo? Also, here we append, in the previous if clause there's cat with replace.